lib/puppet/provider/augeasprovider/default.rb

```diff
@@ -162,6 +162,10 @@ def self.augsave!(aug, reload = false)
     raise Augeas::Error, 'Failed to save Augeas tree to file. See debug logs for details.'
   ensure
     aug.load! if reload
+    # https://github.com/hercules-team/augeas/commit/eb04250a05671b2d001444b72b8778328d209d75 introduced a bug in libaugeas.so.0.25.0
+    # bundled with puppet7.
+    # A temporary fix is to invoke load! twice.
+    aug.load! if reload && Puppet::Util::Package.versioncmp(Puppet.version, '7.0.0') >= 0
```

Comment:
This depends on the Augeas version, not Puppet. So I'd prefer to use `aug_version` instead. So something like:
```suggestion
    aug.load! if reload && Puppet::Util::Package.versioncmp(aug_version, '1.13.0') >= 0
```

Reply:
Hmm, I tried this but then my test broke ....
`puts "Puppet version: #{Puppet.version}, Augeas version: #{aug_version}"` says

```
Puppet version: 6.29.0, Augeas version: 1.12.0
Puppet version: 7.24.0, Augeas version: 1.12.0
```

Reply:
Oh that's weird. On my Fedora with system Ruby:
```console
$ pry
[1] pry(main)> require 'augeas'
=> true
[2] pry(main)> Augeas.open(nil, nil, Augeas::NO_MODL_AUTOLOAD) { |aug| aug.get('/augeas/version') }
=> "1.13.0"
```

On my EL8 AIO:

```console
$ rpm -qv puppet-agent
puppet-agent-7.24.0-1.el8.x86_64
$ /opt/puppetlabs/puppet/bin/ruby 
require 'augeas'
puts Augeas.open(nil, nil, Augeas::NO_MODL_AUTOLOAD) { |aug| aug.get('/augeas/version') }

1.13.0
```

On a Debian 11 machine with `puppet-agent-7.24.0-1bullseye`:
```console
# /opt/puppetlabs/puppet/bin/ruby
require 'augeas'
puts Augeas.open(nil, nil, Augeas::NO_MODL_AUTOLOAD) { |aug| aug.get('/augeas/version') }
1.13.0
```

Reply:
Ah, unit tests does not use puppet agent, and on my rockylinux-8 vagrant box augeas-lib 1.12.0 is installed.
Should I just drop the unit test I added then?

Reply:
Figured out how to stub it :)